packages/neon_framework/packages/account_repository/test/account_repository_test.dart

```diff
@@ -0,0 +1,584 @@
+      });
+
+      test('uses initialAccount as active', () async {
+        when(() => storage.readAccounts()).thenAnswer((_) async => credentialsList);
+        await repository.loadAccounts(initialAccount: credentialsList[1].id);
+
+        expect(
+          repository.accounts,
+          emits(
+            isA<_AccountStream>()
+                .having((e) => e.accounts, 'accounts', containsAllInOrder(accountsList))
+                .having((e) => e.active, 'active', equals(accountsList[1])),
+          ),
+        );
+
+        verifyNever(() => storage.readLastAccount());
+      });
+
+      test('ignores initialAccount when not it accounts', () async {
+        when(() => storage.readAccounts()).thenAnswer((_) async => credentialsList);
+        await repository.loadAccounts(initialAccount: 'invalid');
+
+        expect(
+          repository.accounts,
+          emits(
+            isA<_AccountStream>()
+                .having((e) => e.accounts, 'accounts', containsAllInOrder(accountsList))
+                .having((e) => e.active, 'active', equals(accountsList.first)),
+          ),
+        );
+
+        verifyNever(() => storage.readLastAccount());
+      });
+
+      test('uses last active account when specified', () async {
+        when(() => storage.readAccounts()).thenAnswer((_) async => credentialsList);
+        when(() => storage.readLastAccount()).thenAnswer((_) async => credentialsList[1].id);
+        await repository.loadAccounts(rememberLastUsedAccount: true);
+
+        expect(
+          repository.accounts,
+          emits(
+            isA<_AccountStream>()
+                .having((e) => e.accounts, 'accounts', containsAllInOrder(accountsList))
+                .having((e) => e.active, 'active', equals(accountsList[1])),
+          ),
+        );
+
+        verify(() => storage.readLastAccount()).called(1);
+      });
+
+      test('ignores last active account when not in accounts', () async {
+        when(() => storage.readAccounts()).thenAnswer((_) async => credentialsList);
+        when(() => storage.readLastAccount()).thenAnswer((_) async => 'invalid');
+        await repository.loadAccounts(rememberLastUsedAccount: true);
+
+        expect(
+          repository.accounts,
+          emits(
+            isA<_AccountStream>()
+                .having((e) => e.accounts, 'accounts', containsAllInOrder(accountsList))
+                .having((e) => e.active, 'active', equals(accountsList.first)),
+          ),
+        );
+
+        verify(() => storage.readLastAccount()).called(1);
+      });
+
+      test('prefers last active account over initial account', () async {
+        when(() => storage.readAccounts()).thenAnswer((_) async => credentialsList);
+        when(() => storage.readLastAccount()).thenAnswer((_) async => credentialsList[1].id);
+        await repository.loadAccounts(
+          rememberLastUsedAccount: true,
+          initialAccount: credentialsList.first.id,
```

Comment:
Should initialAccount rather be called fallbackAccount?
IMO this combination of inputs to the method makes no sense, unless the initialAccount is only a fallback, which it is.
Both parameters are imperative about using a specific account, but it isn't clear which takes precedence.

Reply:
I'm not sure if `fallback` makes sense.
IMO a fallback value is used when no other value is available or the available value is invalid.
In this case the fallback would be the first account available `active = accounts.keys.firstOrNull;`.

Reply:
Right, should we even take it into account then?
Maybe this functionality should be done through switchAccount instead if necessary.

Reply:
Let's leave it for now and see how the other refactors go.
I explicitly treid to mimic the current `AccountsBloc` and where possible would try similar with other repository migrations.
I fear that we'll try to rewrite everything at once.

I can open an issue for this if you want.